src/main/webapp/dev/js/main/instructorHelp.js

```diff
@@ -25,7 +34,224 @@ function bindPanelCollapseLinksAndAnchor() {
+        index.setRef('id');
+        $.each(questionMap, (id, question) => {
+            index.addDoc(question);
+        });
+    }
+
+    $.each($('#topics li'), (idx, li) => {
+        const topicId = $(li).find('a').attr('href');
+        addQuestionsForTopic(topicId);
+    });
+    buildIndex();
+}
+
+function renderPage(page) {
+    if (page < 1 || page > numPages) {
+        return;
+    }
+    if (page === 'prev') {
+        renderPage(curPage - 1);
+        return;
+    }
+    if (page === 'next') {
+        renderPage(curPage + 1);
+        return;
+    }
+
+    curPage = page;
+    const pageNum = page;
+    const searchResultElements = $('#searchResults').children();
+    const pageElements = $(searchResultElements.slice((pageNum - 1)*numResultsPerPage,
+                                                     (pageNum - 1)*numResultsPerPage + numResultsPerPage));
+
+    // hide all search results by default
+    searchResultElements.each(function () {
+        $(this).hide();
+    });
+    // show search results belonging to current page
+    pageElements.each(function () {
+        $(this).show();
+    });
+    // render paging controls
+    $('#pagingControls').children().each((idx, button) => {
+        if (idx === pageNum) {
+            $(button).prop('disabled', true);
+        } else {
+            $(button).prop('disabled', false);
+        }
+    });
+    $('#prevPage').prop('disabled', false);
+    $('#nextPage').prop('disabled', false);
+
+    // disable prev and next buttons for first and last page respectively
+    if (pageNum === 1) {
+        $('#prevPage').prop('disabled', true);
+    }
+    if (pageNum === numPages) {
+        $('#nextPage').prop('disabled', true);
+    }
+}
+
+function searchQuestions() {
+    const query = $('#searchQuery').val();
+
+    $('#searchResults').empty();
+    $('#pagingControls').empty();
+    $('#pagingControls').removeClass('padding-15px margin-bottom-35px');
+    if (query === '') {
+        $('#allQuestions').show();
+        $('#pagingDivider').hide();
+        $('#searchMetaData').text('');
+        return;
+    }
+    const results = index.search(query, {
+        fields: {
+            title: { boost: 4 },
+            tags: { boost: 2 },
+            body: { boost: 0.5 },
+        },
+        bool: 'AND',
+    });
+    $.each(results, (idx, result) => {
+        const { htmlId } = questionMap[result.ref];
+        const questionPanel = $(`#${htmlId}`)[0];
+        $('#searchResults').append(questionPanel.outerHTML);
+    });
+
+    // highlight matches for keyword in the search results
+    function highlightKeyword(keyword) {
+        const highlightRegex = new RegExp(keyword, 'g');
+        $('#searchResults').children().each(function () {
+            // highlight heading matches for query
+            $(this).find('.panel-heading').html(function () {
+                return $(this).html().replace(highlightRegex, `<span class='text-bold color-positive'>${keyword}</span>`);
```

Comment:
When you change the styling, ensure that it contrasts well with both regular text and text in examples (e.g. the following is a negative example)
![screen shot 2018-07-14 at 2 34 19 am](https://user-images.githubusercontent.com/12183503/42703054-af8fbe50-870e-11e8-9a59-6a9578aff879.png)